File: backend/app/schemas/progress.py
from datetime import datetime
from pydantic import BaseModel


class MilestoneItem(BaseModel):
    id: str
    title: str
    description: str
    timestamp: datetime
    type: str  # 'assessment', 'plan_created', 'plan_completed', 'skill_mastered'


class StudentProgressSummary(BaseModel):
    streak_days: int
    plans_completed: int
    skills_mastered: int
    total_tasks: int
    completed_tasks: int
    materials_count: int
    average_skill_score: float
    recent_milestones: list[MilestoneItem]

File: frontend/src/api/progress.ts
import { apiClient } from './client';

export interface MilestoneItem {
  id: string;
  title: string;
  description: string;
  timestamp: string;
  type: 'assessment' | 'plan_created' | 'plan_completed' | 'skill_mastered';
}

export interface StudentProgressSummary {
  streak_days: number;
  plans_completed: number;
  skills_mastered: number;
  total_tasks: number;
  completed_tasks: number;
  materials_count: number;
  average_skill_score: number;
  recent_milestones: MilestoneItem[];
}

export const progressApi = {
  getSummary: async (): Promise<StudentProgressSummary> => {
    const response = await apiClient.get<StudentProgressSummary>('/student/progress');
    return response.data;
  },
};
